resources/chain.rb

```diff
@@ -1,72 +1,64 @@
 include Iptables::Cookbook::Helpers
 
-property :table, [Symbol, String],
-          equal_to: [:filter, :mangle, :nat, :raw, :security, 'filter', 'mangle', 'nat', 'raw', 'security'],
-          default: :filter,
-          description: 'The table the chain should exist on'
-
-property :chain, [Symbol, String],
-          description: 'The name of the Chain'
-
-property :value, String,
-          default: 'ACCEPT [0:0]',
-          description: 'The default action and the Packets : Bytes count'
+property :config_file, String,
+          default: lazy { default_iptables_rules_file(ip_version) },
+          description: 'The full path to find the rules on disk'
 
-property :ip_version, Symbol,
-          equal_to: %i(ipv4 ipv6),
-          default: :ipv4,
-          description: 'The IP version, 4 or 6'
+property :owner, String,
+          default: 'root',
+          description: 'Permissions on the saved output file'
 
-property :file_mode, String,
-          default: '0644',
+property :group, String,
+          default: 'root',
           description: 'Permissions on the saved output file'
 
-property :source_template, String,
-          default: 'iptables.erb',
-          description: 'Source template to use to create the rules'
+property :mode, String,
+          default: '0600',
+          description: 'Permissions on the saved output file'
 
 property :cookbook, String,
           default: 'iptables',
           description: 'Source cookbook to find the template in'
 
+property :template, String,
+          default: 'iptables.erb',
+          description: 'Source template to use to create the rules'
+
 property :sensitive, [true, false],
-          default: false,
-          description: 'mark the resource as senstive'
+          default: true,
+          description: 'Mark the resource as senstive'
 
-property :config_file, String,
-          default: lazy { default_iptables_rules_file(ip_version) },
-          description: 'The full path to find the rules on disk'
+property :ip_version, [Symbol, String],
+          coerce: proc { |p| p.to_sym },
+          equal_to: %i(ipv4 ipv6),
+          default: :ipv4,
+          description: 'The IP version, 4 or 6'
 
-action :create do
-  # We are using the accumalator pattern here
-  # This is as we are managing a single config file but using multiple
-  # resouces to allow a cleaner api for the end user
-  # Note, this will only ever go as a file on disk at the end of a chef run
-  table = convert_to_symbol_and_mark_deprecated('table', new_resource.table) if new_resource.table
-  chain = convert_to_symbol_and_mark_deprecated('chain', new_resource.chain) if new_resource.chain
+property :table, [Symbol, String],
+          coerce: proc { |p| p.to_sym },
+          equal_to: Iptables::Cookbook::Helpers::IPTABLES_TABLE_NAMES,
+          required: true,
+          description: 'The table the chain should exist on'
```

Comment:
> 
> 
> This appears not to respect the `cookbook` property and will always use the template from this cookbook instead of a specified one.
> 
> Do we want to keep support for C6 since it is EOL?

Hmm i'll check this out but it should do, the only change that will effect it is it'll take the setting from the _first_ resource called (service/chain/rule) as I only initialise the template resource if it doesn't already exist. Might refactor that thinking about it but then it'll come down the _last_ resource declaration so i'm not 100% convinced it's actually better either.